src/sagemaker_algorithm_toolkit/hyperparameter_validation.py

```diff
@@ -244,15 +244,26 @@ def _visit(name, visited, stack):
         return dependencies_stack
 
     def validate(self, user_hyperparameters):
-        # NOTE: 0. Validate required or fill in default.
+        # NOTE: 0. Convert possible aliases to original terms
+        tmp_user_hyperparameters = {}
+        hp_aliases = {"learning_rate": "eta", "min_split_loss": "gamma", "reg_lambda": "lambda", "reg_alpha": "alpha"}
+
+        for hp, value in user_hyperparameters.items():
+            if hp in hp_aliases:
+                tmp_user_hyperparameters[hp_aliases[hp]] = value
+            else:
+                tmp_user_hyperparameters[hp] = value
+        user_hyperparameters = tmp_user_hyperparameters
+
+        # NOTE: 1. Validate required or fill in default.
```

Comment:
This approach seems to handle aliases correctly. As an alternative, how about creating a method such as `_declare_alias(key_name, alias_name)` and use that to declare aliases when initializing hyperparametes? It may help with readability since aliases will be declared closer to where hps are initialized (as opposed to maintaining a list within the `validate` method itself). Similarly, moving logic to `_declare_alias()` will help with keeping code modular. 

What do you think?